rust/src/schema/arrow_convert.rs

```diff
@@ -8,9 +8,13 @@ use arrow::datatypes::{
 use arrow::error::ArrowError;
 use lazy_static::lazy_static;
 use regex::Regex;
+use std::collections::HashMap;
 use std::convert::TryFrom;
 use std::sync::Arc;
 
+/// Column metadata that stores JSON serialized Arrow DataType
+pub const ARROW_TYPE_METADATA_KEY: &str = "ARROW:schema";
```

Comment:
this is the same key that parquet arrow writer uses, could change to a `delta.` prefixed key but wanted to get buy in as those are reserved.

Reply:
Hmm if we are going to use that key, could we instead make it at the schema level (not the field level) and use the same format as is in Parquet?

As described [here](https://arrow.apache.org/docs/cpp/parquet.html#roundtripping-arrow-types):

> The Arrow schema is serialized as a [Arrow IPC](https://arrow.apache.org/docs/format/Columnar.html#format-ipc) schema message, then base64-encoded and stored under the ARROW:schema metadata key in the Parquet file metadata.

Reply:
I could change the key name.

I went this route because I don't see metadata on delta's SchemaTypeStruct. And used JSON as the metadata values on Field are serde_json::Value's